system/jlib/jthread.cpp

```diff
@@ -837,8 +813,8 @@ void CAsyncFor::For(unsigned num,unsigned maxatonce,bool abortFollowingException
             }
         }
     }
-    if (e)
-        throw e;
+    if (e.isSet())
```

Comment:
should AtomicShared have an operatr * (like Owned/Shared) so can just say if (e) ?

Reply:
I think it does - I'll check

Reply:
I think there isn't a operator * because it isn't generally thread-safe - only if it was linked before returning, and even that isn't thread sfae (hence the getLinkNonAtomic).